orthopairs/src/main/java/org/reactome/release/orthopairs/Main.java

```diff
@@ -0,0 +1,155 @@
+     *
+     *  At the end, we have the same two files for each species: The Gene-Protein mapping file specific to the species, and the Protein Homolog file.
+     *  All Protein IDs are UniProt (instead of the mix of Ensembl and UniProt) and all Gene IDs match the format used by Ensembl.
+     *
+     *  These files are much smaller than the older version, since PANTHER annotates which homologs are the 'Least Diverged', meaning we can filter out many that add noise to the dataset.
+     *
+     */
+
+    private static final Logger logger = LogManager.getLogger();
+
+    public static void main( String[] args ) throws IOException, ParseException {
+
+        String pathToConfig = "";
+        String sourceMappingSpecies = "";
+        // If using an alternative source species, specify the 4-letter code as the second argument
+        //TODO: Better solution for this. What if user wants default config but to change source species?
+        if (args.length > 0) {
+            pathToConfig = args[0];
+            sourceMappingSpecies = args[1];
+        } else {
+            pathToConfig = "src/main/resources/config.properties";
+            sourceMappingSpecies = "hsap";
+        }
+        Properties props = new Properties();
+        props.load(new FileInputStream(pathToConfig));
+        // Load all config properties
+        String releaseNumber = props.get("releaseNumber").toString();
+        String pathToSpeciesConfig = props.get("pathToSpeciesConfig").toString();
+        String pantherFilepath = props.get("pantherFileURL").toString();
+        String pantherQfOFilename = props.get("pantherQfOFilename").toString();
+        String pantherHCOPFilename = props.get("pantherHCOPFilename").toString();
+        String MGIFileURL = props.get("MGIFileURL").toString();
+        String RGDFileURL = props.get("RGDFileURL").toString();
+        String XenbaseFileURL = props.get("XenbaseFileURL").toString();
+        String ZFINFileURL = props.get("ZFINFileURL").toString();
+
+        new File(releaseNumber).mkdir();
+
+        logger.info("Starting Orthopairs file generation");
+        // Download and extract homology files from Panther
+        List<String> pantherFiles = new ArrayList<String>(Arrays.asList(pantherQfOFilename, pantherHCOPFilename));
+        for (String pantherFilename : pantherFiles) {
+            downloadAndExtractTarFile(pantherFilename, pantherFilepath);
+        }
+
+        // Download ID files from various model organism databases (Mouse Genome Informatics, Rat Genome Database, Xenbase (frog), ZFIN (Zebrafish))
+        // HGNC identifier file is downloaded as well.
+        List<String> alternativeIdMappingURLs = new ArrayList<>(Arrays.asList(MGIFileURL,RGDFileURL,XenbaseFileURL,ZFINFileURL));
+        for (String altIdURL : alternativeIdMappingURLs) {
+            File altIdFilepath = new File(altIdURL.substring(altIdURL.lastIndexOf("/")+1));
```

Comment:
Something like this might be more declarative: `Paths.get(altIdURL).getFileName().toFile())`.

You may need to use `new URI(altIdURL)` for the URL to be handled properly, but the Paths.get() method may be smart enough to use the URL as a String.  We'd need to try it out.